common/src/main/java/org/apache/sedona/common/Functions.java

```diff
@@ -70,6 +70,27 @@ public static double length(Geometry geometry) {
         return geometry.getLength();
     }
 
+    public static Double x(Geometry geometry) {
```

Comment:
Does this need to be the boxed type instead of the native double like the other functions return?

Reply:
ST_X, ST_Y and ST_Z should return null if the first argument is not a Point, so we have to check the input type somewhere. I chose to do that in the common functions, because if they can direcly return null by using a boxed type, we can simply wrap it with InferredUnaryExpression in the Spark SQL functions. But as you pointed out, returning boxed type looks somewhat strange. Should I change the common functions to return a primitive value? In that case, we have to use UnaryGeometryExpression instead of InferredUnaryExpression so the code will become a bit verbose. Either is fine with me.

Reply:
Right that makes sense then, nvm